devlib/instrument/acmecape.py

```diff
@@ -81,27 +82,41 @@ def start(self):
         self.process = Popen(self.command.split(), stdout=PIPE, stderr=STDOUT)
 
     def stop(self):
-        self.process.terminate()
+        still_running = True
+        self.logger.info('Terminating iio-capture [%d] instance',
+                         self.process.pid)
         timeout_secs = 10
         output = ''
+        # Gracefully terminate the process (SIGTERM)
         for _ in xrange(timeout_secs):
+            self.process.terminate()
+            time.sleep(1)
             if self.process.poll() is not None:
+                still_running = False
                 break
-            time.sleep(1)
         else:
             output += _read_nonblock(self.process.stdout)
-            self.process.kill()
-            self.logger.error('iio-capture did not terminate gracefully')
-            if self.process.poll() is None:
+            # Kill a non responing process (SIGKILL)
+            for _ in xrange(timeout_secs):
+                self.process.kill()
+                time.sleep(1)
+                if self.process.poll() is not None:
+                    still_running = False
+                    break
+            else:
+                self.logger.error('iio-capture did not terminate gracefully')
                 msg = 'Could not terminate iio-capture:\n{}'
                 raise HostError(msg.format(output))
-        if self.process.returncode != 15: # iio-capture exits with 15 when killed
+
+        # iio-capture returns the signal sent to stop it
+        if still_running:
```

Comment:
Can this condition ever be reached with the value being `True`?
Also this does not seem be checking for other error codes being returned anymore if the process has terminated?